components/logins/src/login.rs

```diff
@@ -490,37 +509,138 @@ impl Login {
 
             time_password_changed: row.get("timePasswordChanged")?,
             times_used: row.get("timesUsed")?,
+            enc_fields: row.get("encFields")?,
         };
-        // For now, we want to apply fixups but still return the record if
-        // there is unfixably invalid data in the db.
-        Ok(login.maybe_fixup().unwrap_or(None).unwrap_or(login))
+        // XXX - we used to perform a fixup here, but that seems heavy-handed
+        // and difficult - we now only do that on add/insert when we have the
+        // encryption key.
+        Ok(login)
+    }
+}
+
+impl ValidateAndFixup for Login {
+    fn validate_and_fixup(&self, fixup: bool) -> Result<Option<Self>> {
+        // We don't validate or fixup metadata, which we arguably should.
+        Ok(self
+            .fields
+            .validate_and_fixup(fixup)?
+            .map(|new_fields| Self {
+                id: self.id.clone(),
+                fields: new_fields,
+                enc_fields: self.enc_fields.clone(),
+                time_created: self.time_created,
+                time_password_changed: self.time_password_changed,
+                time_last_used: self.time_last_used,
+                times_used: self.times_used,
+            }))
+    }
```

Comment:
I'm not sure if this should be implemented on the `Login` struct since there's no way to validate the encrypted fields.  It seems better to validate the `LoginFields`, then decrypt `enc_fields` and use the `EncryptedLogin.validate_and_fixup()` code.

Reply:
hrm, yeah, that's probably correct.